src/planner/binder/query_node/plan_select_node.cpp

```diff
@@ -97,21 +98,33 @@ unique_ptr<LogicalOperator> Binder::CreatePlan(BoundSelectNode &statement) {
 
 	for (size_t i = 0; i < statement.modifiers.size(); i++) {
 		auto &modifier = statement.modifiers[i];
-		if (modifier->type != ResultModifierType::LIMIT_MODIFIER) {
-			continue;
-		}
-		auto &limit_modifier = (BoundLimitModifier &)*modifier;
-		if (limit_modifier.limit || limit_modifier.offset) {
-			PlanSubqueries(&limit_modifier.limit, &root);
-			PlanSubqueries(&limit_modifier.offset, &root);
-			auto limit = make_unique<LogicalLimit>(limit_modifier.limit_val, limit_modifier.offset_val,
-			                                       move(limit_modifier.limit), move(limit_modifier.offset));
-			limit->AddChild(move(root));
-			root = move(limit);
-			// Delete from modifiers
-			std::swap(statement.modifiers[i], statement.modifiers.back());
-			statement.modifiers.erase(statement.modifiers.end() - 1);
-			i--;
+		if (modifier->type == ResultModifierType::LIMIT_MODIFIER) {
+			auto &limit_modifier = (BoundLimitModifier &)*modifier;
+			if (limit_modifier.limit || limit_modifier.offset) {
+				PlanSubqueries(&limit_modifier.limit, &root);
+				PlanSubqueries(&limit_modifier.offset, &root);
+				auto limit = make_unique<LogicalLimit>(limit_modifier.limit_val, limit_modifier.offset_val,
+				                                       move(limit_modifier.limit), move(limit_modifier.offset));
+				limit->AddChild(move(root));
+				root = move(limit);
+				// Delete from modifiers
+				std::swap(statement.modifiers[i], statement.modifiers.back());
+				statement.modifiers.erase(statement.modifiers.end() - 1);
+				i--;
+			}
+		} else if (modifier->type == ResultModifierType::LIMIT_PERCENT_MODIFIER) {
+			auto &limit_modifier = (BoundLimitPercentModifier &)*modifier;
+			if (limit_modifier.limit) {
+				PlanSubqueries(&limit_modifier.limit, &root);
```

Comment:
PlanSubqueries on the `offset` also needs to happen here.

Reply:
Great, it is the part I missed. I will fix it. Thanks.